### PR TITLE
CHUI-77: Fix mandatory invoice address

### PR DESCRIPTION
**Pull request by jeffersontuc**
#### What is the purpose of this pull request?

This PR fixes a bug, that happens when the invoice is active, which would allow the user to proceed to the payment step without having inserted a valid invoice address.

Now, the "go to payment" button is only present when a proper invoice address is inserted.

#### How should this be manually tested?

- Use this [workspace](https://jeff--vtexgame1invoice.myvtex.com/checkout/cart/add/?sku=297&qty=1&seller=1&sc=1)
- Proceed to the Omnishipping
- Search for pickup points and use `22250-040` as the postal code
- Pick `Loja em Copacabana`
- Click on "Preencher endereço de nota fiscal"
- You shouldn't see the "go to payment" button until the invoice address is properly inserted.

#### Screenshots or example usage

image

#### Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

**Comment by jeffersontuc** [comment]
I linked these changes in the workspace `beta` and I noticed some tests started to fail. I'm gonna investigate it.

**Comment by jeffersontuc** [comment]
> I linked these changes in the workspace `beta` and I noticed some tests started to fail. I'm gonna investigate it.

I already found the problem and commited the necessary changes.

**Comment by kevinch** [comment]
@jeffersontuc small ui issue when loading. Seems a width issue.

(sorry for massive screenshot)

Screen Shot 2020-06-23 at 09 57 17 (2)

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Fixed
+
+- Now the "go to payment" button, when the invoice is active, is only present when there's a valid invoice address.
```
Comment (kevinch) [inline CHANGELOG.md:12]:
```suggestion
- "go to payment" button's presence only on valid invoice address.
```

**Comment by jeffersontuc** [comment]
> @jeffersontuc small ui issue when loading. Seems a width issue.
> 
> (sorry for massive screenshot)
> 
> Screen Shot 2020-06-23 at 09 57 17 (2)

Yeah. I just noticed it too. I'm working on it.

File: react/utils/Address.js [hunk 2]
```diff
@@ -29,3 +29,14 @@ export const addressHasValidGeocoordinates = address => {
...
+export const isValidAddress = (address, rules) => {
+  const validatedAddress = validateAddress(address, rules)
+
+  return (
+    validatedAddress.city.valid &&
+    validatedAddress.street.valid &&
+    validatedAddress.number.valid &&
+    validatedAddress.neighborhood.valid
+  )
+}
```
Comment (marcoskwkm) [inline react/utils/Address.js:42]: Omnishipping's `/utils` folder [will die](https://vtex-dev.atlassian.net/browse/CHUI-79?atlOrigin=eyJpIjoiZTE3NDc5NTUzNTg1NGRkODliZTU4YTQ2OWI1ZWE5ODciLCJwIjoiaiJ9). Move this to shipping-manager.

Comment (jeffersontuc) [reply react/utils/Address.js:42]: gotcha!

**Comment by jeffersontuc** [comment]
I've commited a fix for the tests which were failing.

**Comment by jeffersontuc** [comment]
Rebased

**Pull request merged**